editor/src/components/inspector/inspector-strategies/inspector-strategies.tsx

```diff
@@ -136,12 +138,29 @@ export const setPropFillStrategies = (axis: Axis): Array<InspectorStrategy> => [
 
     return elements.flatMap((path) => {
       const parentInstance = MetadataUtils.findElementByElementPath(metadata, EP.parentPath(path))
-      return MetadataUtils.isFlexLayoutedContainer(parentInstance)
-        ? [
-            setProperty('always', path, PP.create(['style', 'flexGrow']), '1'),
-            nukePropsCommand(path),
-          ]
-        : [setProperty('always', path, PP.create(['style', widthHeightFromAxis(axis)]), '100%')]
+      if (!MetadataUtils.isFlexLayoutedContainer(parentInstance)) {
+        return [
+          nukePropsCommand(path),
+          setProperty('always', path, PP.create(['style', widthHeightFromAxis(axis)]), '100%'),
+        ]
+      }
+
+      const flexDirection = detectFlexDirectionOne(metadata, EP.parentPath(path)) ?? 'row'
+
+      if (
+        (flexDirection.startsWith('row') && axis === 'vertical') ||
+        (flexDirection.startsWith('column') && axis === 'horizontal')
+      ) {
+        return [
+          nukePropsCommand(path),
```

Comment:
is there an importance to the order of nukePropsCommand and setProperty? here nukePropsCommand comes first, on line 162, nukePropsCommand comes last

Reply:
the order does matter in the general case, because these commands are passed to [`foldAndApplyCommandsSimple`](https://github.com/concrete-utopia/utopia/blob/82c5361ce43233c4eda86b2b25d0dc551db5921c/editor/src/components/canvas/commands/commands.ts#L177), so they are applied sequentially.

However, in these two specific cases the order is OK, because
- in the place where you commented, the width/height prop is removed first and then it is readded as the last step,
- in the other branch, the width/height prop is removed, and flex-grow is added (which are orthogonal, so the order doesn't matter).